src/components/Account-panel/Delivery-address/Delivery-address.tsx

```diff
@@ -1,3 +1,233 @@
+        icon = <IconNovaPoshta style={stIconM} />
+        addressHeadline = 'Нова пошта'
+        break
+      case 'ukr_poshta':
+        icon = <IconUkrPoshta style={stIconM} />
+        addressHeadline = 'Укрпошта'
+        location = 'Відділення'
+        titleAddress = `${event.city}, ${event.street}, ${event.region}, ${event.post_code}, буд.${event.house_number}, кв.${event.apartment_number}`
+        break
+      default:
+        break
+    }
+  }
+
+  const setLocationAndAddress = () => {
+    if (addressType === 'адреса') {
+      location = 'Адреса'
+      titleAddress = `${event.city}, ${event.street}, буд.${event.house_number}, кв.${event.apartment_number}`
+    } else if (addressType === 'відділення') {
+      location = determineLocation(addressWarehouse ?? undefined)
+      titleAddress = `${location} ${event.address_warehouse}, ${event.city}`
+    }
+  }
+
+  setIconAndHeadline(source)
+  setLocationAndAddress()
+
+  return {
+    icon,
+    text: type,
+    addressHeadline,
+    location,
+    titleAddress,
+    address_warehouse: addressWarehouse
+  }
+}
+
+const determineLocation = (addressWarehouse?: string): string => {
+  if (!addressWarehouse) return ''
+  return addressWarehouse.startsWith('#') ? 'Поштомат' : 'Відділення'
+}
+
 export const DeliveryAddress = () => {
-  return <div>DeliveryAddress</div>
+  const [deliveryAddresses, setDeliveryAddresses] = useState({
+    nova_poshta: [],
+    ukr_poshta: []
+  })
+  const [snackbarData, setSnackbarData] = useState<ISnackbarData>({
+    open: false,
+    error: false
+  })
+  const handleCloseSnackbar = () => {
+    setSnackbarData({ ...snackbarData, open: false })
+  }
+  const handleDeleteClick = (id: number, source: string) => {
+    let url = ''
+    let dataId = {}
+    if (source === 'nova_poshta') {
+      dataId = { nova_poshta_id: id }
+      url = '/api/posts/remove_nova_poshta_data'
+    } else if (source === 'ukr_poshta') {
+      dataId = { ukr_poshta_id: id }
+      url = '/api/posts/remove_ukr_postal_office'
+    }
+
+    axiosInstance
+      .delete(url, { data: dataId })
+      .then(() => {
+        fetchDataMyPostOffices(setDeliveryAddresses)
+        setSnackbarData({
+          open: true,
+          error: false,
+          message: 'Ваші зміни видалено!'
+        })
+      })
+      .catch((error) => {
+        console.error('Error', error)
+        setSnackbarData({ open: true, error: true, message: 'Сталась помилка' })
+      })
+  }
+
+  useEffect(() => {
+    fetchDataMyPostOffices(setDeliveryAddresses)
+  }, [setDeliveryAddresses])
+
+  const novaPoshtaArray: AddressArray =
+    deliveryAddresses.nova_poshta.map((item: AddressItem) => ({
+      ...item,
+      source: 'nova_poshta',
+      addressType: item.address_warehouse ? 'відділення' : 'адреса'
+    })) || []
+  const ukrPoshtaArray: AddressArray =
+    deliveryAddresses.ukr_poshta.map((item: AddressItem) => ({
+      ...item,
+      source: 'ukr_poshta'
+    })) || []
```

Comment:
why do you need that?

Reply:
I use this in 4 different places, so I took it out separately to make it easier to swipe and shorten the code